ui/src/app/base/components/Meter/Meter.tsx

```diff
@@ -0,0 +1,171 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React, { useEffect, useRef, useState } from "react";
+
+import { useOnWindowResize } from "app/base/hooks";
+
+const DEFAULT_COLORS = ["#0066CC", "#0E8420", "#C7162B", "#F99B11"];
+
+const updateWidths = (
+  el: React.MutableRefObject<Element | null>,
+  maximum: number,
+  setBarWidth: (size: number) => void,
+  setSegmentWidth: (size: number) => void
+) => {
+  const boundingWidth = el?.current?.getBoundingClientRect()?.width || 0;
+  // Because we're dealing with single pixel separators, we set the bar width to
+  // the nearest floor base 2 number to help with anti-aliasing.
+  const base2Width = Math.pow(
+    2,
+    Math.floor(Math.log(boundingWidth) / Math.log(2))
+  );
+  const segmentWidth = base2Width > maximum * 2 ? base2Width / maximum : 2;
+  setBarWidth(base2Width);
+  setSegmentWidth(segmentWidth);
+};
+
+type MeterDatum = {
+  color?: string;
+  key: string;
+  label?: string;
+  value: number;
+};
+
+type Props = {
+  className?: string;
+  data: MeterDatum[];
+  emptyColor?: string;
+  labelsClassName?: string;
+  max?: number;
+  overColor?: string;
+  segmented?: boolean;
+  separatorColor?: string;
+  small?: boolean;
+};
+
+const Meter = ({
+  className,
+  data,
+  emptyColor = "#D3E4ED",
```

Comment:
Yep I can add some consts! As for the naming I think keeping it as "color" makes sense because it reflects how inline styles (`background-color`, `border-color`)